services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/modules/nodeports.py

```diff
@@ -270,33 +293,38 @@ async def _get_data_from_port(
             transferred_bytes = downloaded_file.stat().st_size
 
             # in case of valid file, it is either uncompressed and/or moved to the final directory
-            with log_context(logger, logging.DEBUG, "creating directory"):
+            with log_context(_logger, logging.DEBUG, "creating directory"):
                 final_path.mkdir(exist_ok=True, parents=True)
             port_data = f"{final_path}"
             dest_folder = PrunableFolder(final_path)
 
             if _is_zip_file(downloaded_file):
                 # unzip updated data to dest_path
-                logger.debug("unzipping %s", downloaded_file)
-                unarchived: set[Path] = await unarchive_dir(
-                    archive_to_extract=downloaded_file,
-                    destination_folder=final_path,
-                    progress_bar=sub_progress,
-                )
-
-                dest_folder.prune(exclude=unarchived)
+                _logger.debug("unzipping %s", downloaded_file)
+                try:
+                    unarchived: set[Path] = await unarchive_dir(
+                        archive_to_extract=downloaded_file,
+                        destination_folder=final_path,
+                        progress_bar=sub_progress,
+                    )
+                    dest_folder.prune(exclude=unarchived)
+
+                    _logger.debug("all unzipped in %s", final_path)
+                except ArchiveError:
+                    _logger.warning(
```

Comment:
why is this not an error?

Reply:
so in the issue you refer to I see:
"Expected behaviour
Either: The file is successfully unzipped
Or: A verbose and clear error is presented to the user, suggesting she may use a different zipping algorithm"

How is a warning in docker logs going to cover this?

As I understand the user will be presented with a zip file instead of an unzipped one:
- confusion, suddenly a service behaves differently than the others
- where is the clear and verbose error popping up?
- what should I do as a user to change it?
- what happens if we somehow fix the unzipping and suddenly my projects are broken again because I don't have a zip file anymore?

Reply:
And I still don't see why this is a warning. this should be an error. 
We fail unarchiving. BTW I guess Ruff will not like that this exception does not end with Error

Reply:
So you comprehension is correct.

- We now have different behaviours. The main advantage being their service now opens.
- We should also properly inform the user. For this I would require for the notifications service to exist in the system. This way I can send a notification and a service will properly format it for me with the "label" used by the user in the frontend instead of the node_id. They will also be able to see a warning hovering message about it in the UI, which will not be hidden in the logger.
- You are saying this should still be an error even if we do something since we should fix it? Now that I think about it, maybe we should not even print a message in warning but info, since a clear action will be taken like informing the user.

Reply:
well now I can see what will happen:
- someone uploads a zip file to the filepicker,
- connects the filepicker to a service,
- and sometimes it will unzip, and sometimes not
- That is the worst that can happen no? I can already see the support cases about this.

Is there no way of detecting whether the zip file is unzippable or not?
Currently we do have a sign in the frontend regarding how a file was transmitted between 2 services. Maybe this could be used? at least this would show that something bad happened. this is already wired. maybe that can be something to look into?

Reply:
![image](https://github.com/user-attachments/assets/5730a619-2fa2-41f1-b931-46fcbbb4dace)

Reply:
If you have the file the error is raised when opening it with the zipping tool. So it does not take additional time to check if it's unzippable. It takes time to download.

At this point to avoid different behaviour. For now I can use a tool like `7zip` to unzip, since it appears to support most esoteric forms of archives.

That would be a quick win for this case, where we generate with user generated content.

Reply:
What you are showing in the screenshot is not reliable enough. It is when we start and stop the pulling of the input port.

Also you have no way to distinguish between user generated content and archives created by the dynamic-sidecar